### PR TITLE
modules: add segger module locally, and update to 3.30

### PR DESCRIPTION
**Pull request by nashif**
Move Zephyr related files into modules/ and update to 3.30

### CODE REVIEW

**Comment by carlescufi** [comment]
This seems to fail when using the log RTT backend: 

```
/home/carles/src/zephyr/zephyr/subsys/logging/log_backend_rtt.c: In function 'line_out_drop_mode':
/home/carles/src/zephyr/zephyr/subsys/logging/log_backend_rtt.c:158:6: warning: unused variable 'ret' [-Wunused-variable]
  158 |  int ret = SEGGER_RTT_WriteSkipNoLock(CONFIG_LOG_BACKEND_RTT_BUFFER,
      |      ^~~
/home/carles/src/zephyr/zephyr/subsys/logging/log_backend_rtt.c:162:6: error: 'ret' undeclared (first use in this function)
  162 |  if (ret == 0) {
      |      ^~~
/home/carles/src/zephyr/zephyr/subsys/logging/log_backend_rtt.c:162:6: note: each undeclared identifier is reported only once for each function it appears in
In file included from ../include/sys/util_macro.h:34,
                 from ../include/sys/util.h:17,
                 from ../include/logging/log_msg.h:10,
                 from ../include/logging/log_backend.h:9,
                 from /home/carles/src/zephyr/zephyr/subsys/logging/log_backend_rtt.c:7:
/home/carles/src/zephyr/zephyr/subsys/logging/log_backend_rtt.c: In function 'data_out_block_mode':
/home/carles/src/zephyr/modules/debug/segger/Config/SEGGER_RTT_Conf.h:192:58: error: '_SEGGER_RTT__LockState' undeclared (first use in this function)
  192 |                                                   : "r" (_SEGGER_RTT__LockState)                                 \
      |                                                          ^~~~~~~~~~~~~~~~~~~~~~
```

**Comment by nashif** [comment]
> This seems to fail when using the log RTT backend:

fixed now..

**Comment by wthidden** [comment]
pulled this PR into my working set and still no joy on getting SystemView to work with STM32F429Zi on a nucleo development board. Seems there is no RTT data being sent out.

**Comment by wthidden** [comment]
Updated my repo with latest and now this SystemView works.

**Review by carlescufi — APPROVED** [review]
This works fine now

File: cmake/app/boilerplate.cmake [hunk 1]
```diff
@@ -66,6 +66,8 @@ set_property(GLOBAL PROPERTY GENERATED_KERNEL_SOURCE_FILES "")
 
 set(APPLICATION_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR} CACHE PATH "Application Source Directory")
 set(APPLICATION_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR} CACHE PATH "Application Binary Directory")
+file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/.cmake/api/v1/query")
```
Comment (galak) [inline cmake/app/boilerplate.cmake:69]: Does this have to be here?  Feels like if nothing else a comment about what this is about.

Comment (nashif) [reply cmake/app/boilerplate.cmake:69]: oops, good catch, i was testing something completely unrelated :)

File: modules/segger/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+if(CONFIG_USE_SEGGER_RTT)
+  set(SEGGER_DIR ${ZEPHYR_CURRENT_MODULE_DIR})
+  zephyr_include_directories_ifdef(CONFIG_USE_SEGGER_RTT
+    ${SEGGER_DIR}/SEGGER
+    ${SEGGER_DIR}/Config
+    .
+    )
+  zephyr_sources_ifdef(CONFIG_USE_SEGGER_RTT
```
Comment (tejlmand) [inline modules/segger/CMakeLists.txt:8]: `CONFIG_USE_SEGGER_RTT` is already being checked in line 1.
So no need for the `ifdef` here.

Now the the files are being moved, should we use `zephyr_library()` and `zephyr_library_sources` here, so we don't blindly put those files into `libzephyr.a`.

**Review by galak — APPROVED** [review]

**Pull request merged**